src/Compilers/CSharp/Test/Syntax/Parsing/RawStringLiteralParsingTests.cs

```diff
@@ -0,0 +1,272 @@
+        public void TestReachableSwitchCase2()
+        {
+            CreateCompilation(
+@"class C
+{
+    void M()
+    {
+        switch ("""""" a """""")
+        {
+            case """""""" a """""""":
+                break;
+        }
+    }
+}").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TestUnreachableSwitchCase1()
+        {
+            CreateCompilation(
+@"class C
+{
+    void M()
+    {
+        switch ("""""" a """""")
+        {
+            case """""""" b """""""":
+                break;
+        }
+    }
+}").VerifyDiagnostics(
+                // (8,17): warning CS0162: Unreachable code detected
+                //                 break;
+                Diagnostic(ErrorCode.WRN_UnreachableCode, "break").WithLocation(8, 17));
+        }
+
+        [Fact]
+        public void TestSingleLineRawLiteralInSingleLineInterpolatedString()
+        {
+            CreateCompilation(
+@"class C
+{
+    void M()
+    {
+        var v = $""{""""""a""""""}"";
+    }
+}").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TestSingleLineRawLiteralInMultiLineInterpolatedString1()
+        {
+            CreateCompilation(
+@"class C
+{
+    void M()
+    {
+        var v = $@""{""""""a""""""}"";
+    }
+}").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TestSingleLineRawLiteralInMultiLineInterpolatedString2()
+        {
+            CreateCompilation(
+@"class C
+{
+    void M()
+    {
+        var v = $@""{
+            """"""a""""""
+        }"";
+    }
+}").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TestMultiLineRawLiteralInSingleLineInterpolatedString()
+        {
+            CreateCompilation(
+@"class C
+{
+    void M()
+    {
+        var v = $""{""""""
+
+""""""}"";
+    }
+}").VerifyDiagnostics(
+                // (5,20): error CS9105: Multi-line raw string literals are only allowed in verbatim interpolated strings
+                //         var v = $"{"""
+                Diagnostic(ErrorCode.ERR_Multi_line_raw_string_literals_are_only_allowed_in_verbatim_interpolated_strings, @"""""""
+
+""""""").WithLocation(5, 20));
+        }
+
+        [Fact]
+        public void TestMultiLineRawLiteralInMultiLineInterpolatedString1()
+        {
+            CreateCompilation(
+@"class C
+{
+    void M()
+    {
+        var v = $@""{""""""
+
+""""""}"";
+    }
+}").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TestMultiLineRawLiteralInMultiLineInterpolatedString2()
+        {
+            CreateCompilation(
+@"class C
+{
+    void M()
+    {
+        var v = $@""{
+""""""
+
+""""""
+}"";
+    }
+}").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TestSingleLineRawLiteralContainingClosingBraceInSingleLineInterpolatedString()
+        {
+            CreateCompilation(
+@"class C
+{
+    void M()
+    {
+        var v = $""{""""""}""""""}"";
+    }
+}").VerifyDiagnostics();
+        }
+    }
+}
```

Comment:
Some testing thoughts:
- `IsWhitespace` and `IsNewLine` handle some unusual characters. I'm debating how much of those we should try and cover in tests...
- in terms of trying the raw string literal in various expression contexts, here are some more ideas (based on where `IsPossibleExpression` is used in parser): `is ...`, `is (..., ....)`, `is { Prop: ... }`, in ternary `b ? ... : ...`, in array rank `new int[...]`, in expression statement `...;`, in anon type `new { A = ... }`
- it would be good to test scenario where multiple errors occur in a single string (we only report the first one). For example, multiple lines have the wrong whitespace prefix

Reply:
i was hesitant as it seemed not that important.  in other words, once we've tested a small handful, we know the rest will fall out since hte parser doesn't really do anything special.  that said, if yhou think this is valuable, i have no problem adding.

Reply:
i'll add these.

Reply:
added

Reply:
> is ..., is (..., ....), is { Prop: ... }, in ternary b ? ... : ..., in array rank new int[...], in expression statement ...;, in anon type new { A = ... }

Did you add tests for different contexts?
One more that could be good to include: default parameter values.

Reply:
i did.  but i'll add teh default parameter value case as well.